src/Microsoft.AspNet.Server.Kestrel/Filter/FilteredStreamAdapter.cs

```diff
@@ -28,8 +28,13 @@ public class FilteredStreamAdapter
             _filteredStream = filteredStream;
             _socketInputStream = new SocketInputStream(SocketInput);
 
-            _filteredStream.CopyToAsync(_socketInputStream).ContinueWith((task, state) =>
+            var block = memory.Lease();
+            // Use pooled block for copy
+            _filteredStream.CopyToAsync(_socketInputStream, block).ContinueWith((task, state) =>
             {
+                var returnedBlock = task.Result;
+                returnedBlock.Pool?.Return(returnedBlock);
```

Comment:
If `SocketInput` asks for blocks larger than a pooled block; and then returns them to pool - had a related PR https://github.com/aspnet/KestrelHttpServer/pull/402 to stop it; wasn't sure - too much churn at the time.